### ISSUE TITLE
[Feature]: Pretty-Print JSON (auto-format) without sorting props

### ISSUE THREAD

**Issue by jhonnycano** [sqlitebrowser/sqlitebrowser#3320]
### Describe the new feature

In the "Edit Database Cell" Panel, JSON view mode is very useful, however, the auto-format option sorts alphabetically the properties of the json. While this might be useful in some use cases, in some other scenarios (i.e. debugging and visually finding info, among others), it's better to keep the order of the json unchanged. A settings dialog for this feature would be useful.

**Comment by scottfurry** [comment]
I suspect that little can be done here.

Sorting may be a attribute of the underlying [JSON library](https://github.com/nlohmann/json). I would have to dig into the exact call/usage, I suspect the cell contents are generated from a call to [`nlohmann::basic_json::dump`](https://json.nlohmann.me/api/basic_json/dump/). Internally the library sorts/stores keys alphabetically. I couldn't find a reference for sorting in the API documentation nor could I find a sorting method in the [sqlite documents](https://www.sqlite.org/json1.html).

tl;dr - doubtful that a `no sort` option is possible without interventions that have nothing to do with sqlite.

**Comment by scottfurry** [comment]
We would have to scour the code to see how JSON is being handled to appreciate what is populating the Database Cell. FWIW, that reference lead me to an obscure reference to [JSON object ordering](https://json.nlohmann.me/features/object_order/)

**Comment by justinclift** [comment]
Excellent.  If you've the time and interest, then I reckon try it out and lets see how it goes.  That part of the reason the nightly builds exist. :smile:

**Comment by FrankAnthony11** [comment]
Hi, any news on this? I'm late to the party, but if you save something as json via jsonconvert.serialize with new JsonSerializerSettings { PreserveReferencesHandling = PreserveReferencesHandling.Objects } , and then you open it in the db explorer as pretty json, when it's saved it will format alphabetically properties, then if you try to deserialize later via jsonconvert, it will fail as some $ref will try to get $id that's not yet deserialized because now the order is wrong. and something that had id $2 that used to be at the beginning of the json, is now at the end, and when something before it tried to $ref 2, it will crash

**Comment by justinclift** [comment]
Ahhh.  No-one's working on this presently, as @scottfurry has moved on to other projects.

**Comment by mgrojo** [comment]
This is now implemented thanks to @JPMKTest. @jhonnycano, if you can verify that this is working in the nightly build, please close this issue.
